src/System.CodeDom/src/System/CodeDom/CodeAttributeArgumentCollection.cs

```diff
@@ -0,0 +1,69 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+
+namespace System.CodeDom
+{
+    [Serializable]
+    public class CodeAttributeArgumentCollection : CollectionBase
```

Comment:
Should the CodeDom collections avoid inheriting from `CollectionBase`, to avoid a public API dependency on the non-generic collections? I realize this type of subsetting may not be desirable anymore given the whole .NET Standard 2.0 effort, and given that this is more of a legacy component that mainly exists to help existing code that uses CodeDom to run on .NET Core. But avoiding a dependency on the legacy collections may still be desirable and not inheriting from CollectionBase is unlikely to matter (if at all) for any existing code that uses CodeDom, so felt it was worth asking about. For example, `X509CertificateCollection` inherits from `CollectionBase` in the full framework, but in CoreFX it only inherits from the `IList` interface.